### ISSUE TITLE
Implement iterators support for Collection and Record

### ISSUE THREAD

**Issue by gaperton** [VoliJS/Type-R#44]
https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Symbol/iterator

Record must iterate over all attributes except `undefined`, Collection must iterate over its elements.

Needed to support `for-of` in modern JS.

**Comment by daviderenger** [comment]
This is very welcome! Then we can do async looping with await like
```
for (const model of collection) {
  await model.doStuff();
}
```

Instead of forEach everywhere

**Comment by gaperton** [comment]
Implemented in the upcoming version 3.0 (`develop`)
